exec/java-exec/src/test/java/org/apache/drill/exec/planner/logical/TestConvertCountToDirectScan.java

```diff
@@ -274,4 +274,39 @@ public void testCountsWithWildCard() throws Exception {
       test("drop table if exists %s", tableName);
     }
   }
+
+  @Test
+  public void testCountsForLeafDirectories() throws Exception {
```

Comment:
This test is good.   Were you planning to add one with mix of files and directories since that's a new behavior this PR is adding.

Reply:
Added a unit test for sub-dir with a mix of files and directories.